api/src/main/java/org/apache/iceberg/ContentFile.java

```diff
@@ -165,6 +166,21 @@ default Long fileSequenceNumber() {
    */
   F copyWithoutStats();
 
+  /**
+   * Copies this file with only specific column stats. Manifest readers can reuse file instances;
+   * use this method to copy data and only copy specific stats when collecting files.
+   *
+   * @param columnsToKeepStats the set of the column ids for the columns which stats are kept. If
+   *     the columnsToKeepStats set is empty or <code>null</code>, then all column stats will be
+   *     kept.
+   * @return a copy of this data file, with stats lower bounds, upper bounds, value counts, null
+   *     value counts, and nan value counts for only specific columns.
+   */
+  default F copyWithStats(Set<Integer> columnsToKeepStats) {
```

Comment:
If we decide to do [this](https://github.com/apache/iceberg/pull/8803#discussion_r1375314743) rename, we should probably be consistent here as well.